GitHub Actions: Check URLs using lychee

Calinou opened the pull request:
This makes it possible to be aware of dead links and replace them with other resources (such as archived versions) as soon as possible.

Note that class reference URLs not checked within this repository, as I plan to also integrate lychee into [godotengine/godot](https://github.com/godotengine/godot)'s CI (which has the source of the class reference files).

> [!NOTE]
>
> CI will fail until we resolve https://github.com/godotengine/godot-docs/issues/10674, so that we remove the `download.tuxfamily.org` links from the docs.

Calinou committed:
GitHub Actions: Check URLs using lychee

This makes it possible to be aware of dead links and replace
them with other resources (such as archived versions) as
soon as possible.

tetrapod00 commented on .lycheeignore:
```diff
@@ -0,0 +1,29 @@
+# Mentioned in the class reference.
+.*httpbin\.org\/post
+
+# Class reference metnions `example.com/index.php` in an example, which is 404.
```
```suggestion
# Class reference mentions `example.com/index.php` in an example, which is 404.
```

tetrapod00 reviewed (approved):
Approve the idea. Shader function replacement links are correct.

Can't review the lychee action implementation

skyace65 commented:
I rebased this PR since tetrapod's PR was merged and now the URL check is failing on a different link. I'm not sure why though because the link works for me.

Calinou commented:
> I rebased this PR since tetrapod's PR was merged and now the URL check is failing on a different link. I'm not sure why though because the link works for me.

I've added `gamedevartisan.com` to exclusions. Some websites tend to block link checkers and will return various error codes (typically 429, but sometimes 403 or even 503 in this case).

skyace65 commented:
@mhilbrunner I'd like your review before merging this since it's adding another check

mhilbrunner reviewed (approved):
I think something like this is a good idea, and I didn't spot anything in the workflow. Looks good to me!

Some things can be expanded, but I think we want to see this in action first & iterate. So, lets go ahead.

mhilbrunner commented:
Thank you!
